### PR TITLE
Milestone: React & Redux group project - Space Travelers' Hub

### PR DESCRIPTION
**Pull request by Dangelo-JAN**
# My coding partner and me created a Space Travelers' Hub using React.js and Redux

This app is about a single web app that users can book and join rockets and misssions.

In this milestone we:

- Create React components
- Use React props.
- Use React Router.
- Connect React and Redux.
- Handle events in a React app.
- Write unit tests with React Testing Library.
- Create styles in a React app.
- Use store, actions and reducers in React

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,19 +3,29 @@
...
   "dependencies": {
+    "@reduxjs/toolkit": "^1.8.1",
     "@testing-library/jest-dom": "^5.16.4",
     "@testing-library/react": "^13.2.0",
     "@testing-library/user-event": "^13.5.0",
+    "prop-types": "^15.8.1",
     "react": "^18.1.0",
     "react-dom": "^18.1.0",
+    "react-redux": "^8.0.1",
+    "react-router-dom": "^6.3.0",
```
Comment (mateo951) [inline package.json:14]: [React Redux](https://react-redux.js.org/), [Redux Logger](https://www.npmjs.com/package/redux-logger) and [React Router](https://reactrouter.com/web/guides/quick-start) exist on the package.json file ✔

File: src/App.js [hunk 2]
```diff
@@ -1,25 +1,19 @@
...
-    </div>
-  );
-}
+const App = () => (
+  <div className="App">
+    <NavBar />
+    <Routes>
+      <Route exact path="/" element={<RocketsPage />} />
+      <Route exact path="/missions" element={<MissionsPage />} />
+      <Route exact path="/profile" element={<ProfilePage />} />
```
Comment (mateo951) [inline src/App.js:14]: Proper routes and view components are created properly 🙌

File: src/components/navBar/NavBar.style.scss [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+  &-NavLinks {
+    width: 15rem;
+    display: flex;
+    justify-content: space-between;
+
+    .active {
+      text-decoration: underline;
+    }
+  }
```
Comment (mateo951) [inline src/components/navBar/NavBar.style.scss:40]: The active page should be indicated with an underline ✔

File: src/index.js [hunk 4]
```diff
@@ -1,17 +1,17 @@
...
+import store from './redux/configure-store';
 import App from './App';
-import reportWebVitals from './reportWebVitals';
 
 const root = ReactDOM.createRoot(document.getElementById('root'));
 root.render(
   <React.StrictMode>
-    <App />
-  </React.StrictMode>
+    <Provider store={store}>
```
Comment (mateo951) [inline src/index.js:11]: Redux store is configured properly! Well done 🔥

File: src/redux/rockets/rockets-api.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+const rocketsEndpoint = 'https://api.spacexdata.com/v3/rockets';
```
Comment (mateo951) [inline src/redux/rockets/rockets-api.js:1]: Rockets and Missions data is being fetched successfully from SpaceX API 🤓

File: src/components/MissionsList/MissionsList.jsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        </thead>
+        <tbody>
+          {missions.map((mission) => (
+            <Mission
+              key={mission.mission_id}
+              id={mission.mission_id}
+              name={mission.mission_name}
+              description={mission.description}
+              join={mission.join}
+            />
```
Comment (mateo951) [inline src/components/MissionsList/MissionsList.jsx:36]: Mission component is rendered with the proper info 🔥

File: src/components/JoinedList/JoinedList.jsx [hunk 7]
```diff
@@ -0,0 +1,37 @@
+import { useSelector } from 'react-redux';
+import './JoinedList.styles.scss';
+
+const JoinedList = () => {
+  const rockets = useSelector((state) => state.rockets);
+  const missions = useSelector((state) => state.missions);
```
Comment (mateo951) [inline src/components/JoinedList/JoinedList.jsx:6]: `useSelector()` is used to get a state slice corresponding to routes ✔

File: src/components/Mission/Mission.jsx [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+  } = props;
+  const dispatch = useDispatch();
+
+  const handleClick = () => {
+    if (join) {
+      dispatch(leaveMission(id));
+    } else {
+      dispatch(joinMission(id));
+    }
+  };
```
Comment (mateo951) [inline src/components/Mission/Mission.jsx:20]: "Join Mission" button updates the store successfully by dispatching an action 💪😎 Well done guys ✔

File: src/components/Rocket/Rocket.jsx [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+  const reserveSpan = document.getElementById(id);
+
+  const handleClick = () => {
+    if (reserve) {
+      dispatch(removeRocket(id));
+      reserveSpan.classList.toggle('reserved');
+    } else {
+      dispatch(reserveRocket(id));
+      reserveSpan.classList.toggle('reserved');
+    }
```
Comment (mateo951) [inline src/components/Rocket/Rocket.jsx:24]: "Reserve Rocket" button updates the store successfully by dispatching an action 💪😎 Well done guys ✔

File: src/pages/ProfilePage.jsx [hunk 10]
```diff
@@ -0,0 +1,9 @@
+import JoinedList from '../components/JoinedList/JoinedList';
+
+const ProfilePage = () => (
+  <div className="profile">
+    <JoinedList />
```
Comment (mateo951) [inline src/pages/ProfilePage.jsx:5]: Awesome 🔥 You are rendering a list of reserved rockets and missions without any issue! 

image

File: src/components/JoinedList/JoinedList.jsx [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+
+  const activeRockets = rockets.filter((rocket) => rocket.reserve);
+  const activeMissions = missions.filter((mission) => mission.join);
```
Comment (mateo951) [inline src/components/JoinedList/JoinedList.jsx:9]: The `filter()` method is used to render the reserved rockets and missions 🚀 Fantastic

**Review by mateo951 — APPROVED** [review]
### Hi Team, 

📝👌 **Your project highlights** 😎 
- General Requirements 💪🤓
- App deployed using Netlify 🚀
- Gitflow 🙌
- Data fetch from API's 🤯
- Rendering components properly 🔥

Your project is complete!  There is nothing else to say other than... it's time to merge it 😎 
### Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by hsztan** [comment]
@mateo951 I appreciate your thorough review. 🚀

**Pull request merged**